### ISSUE TITLE
ISBN duplicates problem

### ISSUE THREAD

**Issue by Serenade600** [pcubillos/bibmanager#89]
An edited book has many contributions. Each contribution has its own DOI number, but also shares its parent book's ISBN. Contributions and parent reference are now regarded as duplicates in bibmanager and merge always triggers asking. I have tried to disable exporting ISBN of contributions to bibfile. And this solved the annoyance of asking. But if only focusing on bibfile, this workaround seems lost ISBN information then. Is there a better way to settle this problem down?

**Comment by pcubillos** [comment]
Hi,
Thanks for raising this issue.  Yeah, this is indeed an undesired behavior.  Could you paste a couple of entries that share the same ISBN so that I can start testing for a solution? Thanks!

**Comment by Serenade600** [comment]
Thanks for your fast reply! Here you are.

`@incollection{AAA,
 crossref = {BBB},
 title = {{This is a title of chapter}},
 booktitle = {{This is a title of book}},
 isbn = {9780841212626},
 doi = {10.1021/bk-1989-0406.ch017}
}

@book{BBB,
 title = {{This is a title of book}},
 isbn = {9780841212626},
 doi = {10.1021/bk-1989-0406}
}
`

**Comment by Serenade600** [comment]
I just found that in some edited ebooks each contribution has its own eISBN number. But still, there is ebook that its contributions are not distributed with eISBN. So in this circumstance the parent ISBN maybe stay with them.

**Comment by pcubillos** [comment]
Hi @Serenade600 
OK, I've done some research. As far as I can see, this problem always comes from book-chapter entries, and in all of these cases the entries also contain their respective DOI which differ from each other.

So, the solution during the duplicate detection would be to:
- check ISBNs
- if two ISBNs match, then also check for DOIs

Adding this second step would prevent misidentifying two separate chapters as a same entry (and users wont need to hack into their bibtex entries), while preserving the ISBN duplicate-check for 'regular' books.
This should solve your issues, right?  I'll implement the fix soon then, it should not be much of a problem to do.

**Comment by Serenade600** [comment]
I totally agree with your solution. Thanks for your efforts. In addition, if you could put in your coding source some
```
encoding="utf-8"
```
right behind the right brackets in
```
open(bibfile,****)
```
to encode with UTF-8, which finally becomes
```
open(bibfile,****,encoding="utf-8")
```
, I would much more appreciate it. In last two versions I had to add them myself every time after updating. I am making this suggestion as it does no harm to other functions but only add popularity with little work.

**Comment by Serenade600** [comment]
I totally agree with your solution. Thanks for your efforts. In addition, if you could put in your coding source some
```
encoding="utf-8"
```
right ahead the right brackets in
```
open(bibfile,****)
```
to encode with UTF-8, which finally become
```
open(bibfile,****,encoding="utf-8")
```
, I would much more appreciate it. In last two versions I had to add them myself every time after updating. I am making this suggestion as it does no harm to other functions but only add popularity with little work.

**Comment by Serenade600** [comment]
Anxiously awaiting the fix and hoping that it has not been put off.

**Comment by pcubillos** [comment]
Hi, no, I started but never finished the fix, sorry, my 'real' job got too intense lately.
I'll try to get it done over the weekend.

**Comment by pcubillos** [comment]
OK @Serenade600 
This should be solved in version 1.3.4!
Thanks for pointing this out to me!
